gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -198,6 +200,10 @@ public void launchJob(JobListener jobListener) throws JobException {
             "Previous instance of job %s is still running, skipping this scheduled run", this.jobContext.getJobName()));
       }
 
+      if (this.jobContext.getSemanticsType() == DeliverySemantics.Type.EXACTLY_ONCE) {
+        executeUnfinishedCommitSequences(jobState.getJobName());
```

Comment:
If `executeUnfinishedCommitSequences` throws an exception, the rest of the job flow will be aborted and if the exception is persistent, the job will never make progress.